src/Ledger/Ratify.lagda

```diff
@@ -207,6 +219,27 @@ accepted Γ es@record { cc = cc ; pparams = record { votingThresholds = drepThre
 
 expired : Epoch → GovActionState → Set
 expired current record { expiresIn = expiresIn } = expiresIn <ᵉ current
+
+verifyPrev : (a : GovAction) → NeedsHash a → EnactState → Set
+verifyPrev NoConfidence         h es = let open EnactState es in h ≡ proj₂ cc
+verifyPrev (NewCommittee _ _ _) h es = let open EnactState es in h ≡ proj₂ cc
+verifyPrev (NewConstitution _)  h es = let open EnactState es in h ≡ proj₂ constitution
+verifyPrev (TriggerHF _)        h es = let open EnactState es in h ≡ proj₂ pv
+verifyPrev (ChangePParams _ _)  h es = let open EnactState es in h ≡ proj₂ pparams
+verifyPrev (TreasuryWdrl _)     _ _  = ⊤
+verifyPrev Info                 _ _  = ⊤
+
+delayingAction : GovAction → Bool
+delayingAction NoConfidence         = true
+delayingAction (NewCommittee _ _ _) = true
+delayingAction (NewConstitution _)  = true
+delayingAction (TriggerHF _)        = true
+delayingAction (ChangePParams _ _)  = false
+delayingAction (TreasuryWdrl _)     = false
+delayingAction Info                 = false
+
+delayed : (a : GovAction) → NeedsHash a → EnactState → Bool → Set
+delayed a h es d = ¬ verifyPrev a h es ⊎ d ≡ true
```

Comment:
```suggestion
delayed a h es true = ⊤
delayed a h es _ = ¬ verifyPrev a h es
```
Maybe this would look a bit nicer

Reply:
I actually prefer the current form, but maybe that's something to revisit once we're polishing the PDF